### PR TITLE
ci: add test data and GitHub Actions workflow

### LINKED ISSUE #4: ci: add tests & testing framework
We should have proper testing here; that would include:

- GH Actions CI/CD mechanism setup
- dag/dryrun/integration test for the workflow
- tesing the pipeline at each step of the execution (each rule need separate input test files and corresponding expected output test files)

I know how to set this up; will look into it later.

### PR DESCRIPTION
**Pull request by AngryMaciek**
* add small test dataset to run the pipeline on and compare output against (integration test)
* add GitHub Actions CI mechanism
* add note and shields to README
* add GitHub Issues templates

Fixes #4

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -89,6 +92,8 @@ cat atlas.clusters.2.0.GRCm38.96.bed | sed -E 's/^([0-9]+|[XY])/chr\1/' | sed -E
 # Example for human
 sed -E 's/^([0-9]+|[XYM])/chr\1/' atlas.clusters.2.0.GRCh38.96.bed > atlas.clusters.2.0.GRCh38.96.wchr.bed
 ```
+> Please be mindful of the `biotype_key` key in the configuration file, which should be set according to the annotation type provided to the pipeline.
```
Comment (ninsch3000) [inline README.md]: Great point! Could be moved to line 83 (after first sentence of "Annotations and poly(A) sites") though. Like   
"[...] and gene annotations (e.g. from [ensembl][ensembl]) for your organism and specify their paths in the `config.yaml`. *Please be mindful of the `biotype_key` key in the configuration file, which should be set according to the annotation type provided to the pipeline (e.g. "transcript_biotype" for ensembl gtf files, "transcript_type" for gencode gtf files).*"

**Review by ninsch3000 — CHANGES_REQUESTED** [review]
Thanks a lot again for including the CI @AngryMaciek.
Comments (practically same as in the PAQR repo):
-  is it really necessary to compare every intermediate output file, or could these checks be limited to final output files?
- Could you include a section in the README for a user to test their installation of TPAS? Currently, the integration test is only set up as CI, but maybe a user would want to check whether their installation worked as well.
- other comment is directly in README file

File: README.md (hunk 1 above)
Comment (AngryMaciek) [reply README.md]: Done!

**Comment by AngryMaciek** [comment]
> * is it really necessary to compare every intermediate output file, or could these checks be limited to final output files?

Yeah, I think it is a good practice. We do not need to test every function in each script obviously but I see no reason why would we need to remove the current comparisons from the CI. This is a comp. pipeline so in my opinion having a test (and expected output) for every step (rule) might be highly beneficial. Especially for us - developers, when we ever come back to this repository and decide to change some part of the workflow we could immediately see how the effect of that given change propagates. Also, imagine a situation where you change sth "irrelevant" at the very first step of the pipeline and the end results are unexpectedly different. Tracing back such problems is a pain but with such a checkpoint after every step we have more control and easier insight overall.
All this is good for the CI checks, for the installation check (below) the user would just run the pipeline on the test data without the `cmp` commands.


> * Could you include a section in the README for a user to test their installation of TPAS? Currently, the integration test is only set up as CI, but maybe a user would want to check whether their installation worked as well.

OK, done, I added _Testing the execution_ subsection, one small bash script for the test and run it myself. It runs on the same data as in the CI but without these file checks.

> * other comment is directly in README file

Done

**Comment by AngryMaciek** [comment]
Included comments from @ninsch3000 :

- README update
- `echo` message at the end of the test

I have checked _zarp_ repo and I think that including comparisons in the installation test as in _zarp_ would be too much for me right now...

File: README.md [hunk 2]
```diff
@@ -1,3 +1,6 @@
+[![ci](https://github.com/zavolanlab/tandem-pas/workflows/CI/badge.svg?branch=main)](https://github.com/zavolanlab/tandem-pas/actions?query=workflow%3ACI)
```
Comment (ninsch3000) [inline README.md:1]: This badge shows "no status". Is this desired behaviour?

**Review by ninsch3000 — APPROVED** [review]
Thanks a lot @AngryMaciek !

File: README.md (hunk 2 above)
Comment (AngryMaciek) [reply README.md:1]: For now - yes; URL links to a status on `main`, into which the changes are not merget *yet*.

**Pull request merged**